### PR TITLE
Nemwel blog milestone 2 : Add Data in models

### PR DESCRIPTION
**Pull request by Nemwel-Boniface**
## Hello reviewer(s)
In this milestone I was able to:

- [x] Create model classes for all entities for my program
- [x] Set up associations between models 
- [x] Use models to insert data to my database
- [x] Create custom methods for eg `A method that returns the 3 most recent posts for a given user`
- [x] Check and fix all found offences

Thank you for taking the time to review my project :pray:

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :post, class_name: 'Post', foreign_key: :post_id
+
```
Comment (Mihndim2020) [inline app/models/comment.rb:4]: - [OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_count method every time a new comment has been saved to database, then make the update_comments_count method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :post, class_name: 'Post', foreign_key: :post_id
+
```
Comment (Mihndim2020) [inline app/models/like.rb:4]: - [OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_likes_counter method every time a new like has been saved to database, then make the update_likes_counter method private

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
```
Comment (Mihndim2020) [inline app/models/post.rb:5]: - [OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_count method every time a new post has been saved to database, then make the update_post_count method private

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
### Hi @Nemwel-Boniface, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 1 above)
Comment (Nemwel-Boniface) [reply app/models/comment.rb:4]: Thank you @Mihndim2020 for this suggestion. I will implement this.

**Pull request merged**